public/app/features/search/service/frontend.ts

```diff
@@ -0,0 +1,102 @@
+import { DataFrameView, SelectableValue, ArrayVector } from '@grafana/data';
+import { TermCount } from 'app/core/components/TagFilter/TagFilter';
+
+import { DashboardQueryResult, GrafanaSearcher, QueryResponse, SearchQuery } from '.';
+
+export class FrontendSearcher implements GrafanaSearcher {
+  full = new Map<string, FullResultCache>();
+
+  constructor(private parent: GrafanaSearcher) {}
+
+  async search(query: SearchQuery): Promise<QueryResponse> {
+    if (query.facet?.length) {
+      throw new Error('facets not supported!');
+    }
+    // Don't bother... not needed for this exercise
+    if (query.tags?.length || query.ds_uid?.length) {
+      return this.parent.search(query);
+    }
+
+    // TODO -- make sure we refresh after a while
+    let dash = await this.getDashboards();
+    const view = dash.search(query.query);
+    return {
+      isItemLoaded: () => true,
+      loadMoreItems: async (startIndex: number, stopIndex: number): Promise<void> => {},
+      totalRows: view.length,
+      view,
+    };
+  }
+
+  async getDashboards(): Promise<FullResultCache> {
+    let res = this.full.get('dashboard');
+    if (res) {
+      return Promise.resolve(res);
+    }
+    const v = await this.parent.search({
+      kind: ['dashboard'],
```

Comment:
shouldn't we include folders in the cache?
`      kind: ['dashboard', 'folder'],`